packages/shared/scope-hoisting/src/hoist.js

```diff
@@ -455,23 +464,32 @@ const VISITOR = {
           imported = '*';
         } else if (t.isExportSpecifier(specifier)) {
           imported = specifier.local.name;
+        } else {
+          throw new Error('Unknown export construct');
         }
 
         let id = getIdentifier(asset, 'import', exported.name);
-        asset.symbols.set(exported.name, id.name);
 
         let dep = asset
           .getDependencies()
           .find(dep => dep.moduleSpecifier === source.value);
         if (dep && imported) {
+          let existing = dep.symbols.get(imported);
+          if (existing) {
+            id.name = existing;
+          }
           // this will merge with the existing dependency
           asset.addDependency({
             moduleSpecifier: dep.moduleSpecifier,
             symbols: new Map([[imported, id.name]]),
-            isWeak: true
+            // `symbols` is empty if this dependency was added by JSTransformer/collectDependencies
+            // in that case, override `isWeak`
+            isWeak: dep.symbols.size === 0 ? true : dep.isWeak
```

Comment:
Not sure what the best way to solve this, the behaviour here needs to be:
- If `isWeak` is not set (after the dependency was added by the dependencies collector in JSTransformer), override it with `true`. But `isWeak` is a `boolean`, not `?boolean`.
- otherwise, only set `isWeak` if it already is weak.

I see two possible solutions:
- change `isWeak` to also allow `undefined/null`
- what I did here with the symbols seems fragile